File: maximizeExpression.js
function makeArr(s) {
    let tmp = "";
    let arr = [];
    for (let i = 0; i < s.length; i++) {
        if (s[i] !== "+" && s[i] !== "-" && s[i] !== "*") {
            tmp += s[i];
        } else {
            arr.push(tmp);
            arr.push(s[i]);
            tmp = "";
        }
    }
    arr.push(tmp);
    return [...arr];
}

function foo(oper, e) {
    let arr = makeArr(e);
    for (let i = 0; i < oper.length; i++) {
        for (let j = 1; j < arr.length - 1; j++) {
            let result;
            let flag = 0;
            if (oper[i] === "-" && arr[j] === "-") {
                flag = 1;
                result = +arr[j - 1] - +arr[j + 1];
            }
            if (oper[i] === "+" && arr[j] === "+") {
                flag = 1;
                result = +arr[j - 1] + +arr[j + 1];
            }
            if (oper[i] === "*" && arr[j] === "*") {
                flag = 1;
                result = +arr[j - 1] * +arr[j + 1];
            }
            if (flag === 1) {
                arr[j] = result;
                arr.splice(j - 1, 1);
                arr.splice(j, 1);
                j--;
            }
        }
    }
    return Math.abs(arr[0]);
}

function solution(expression) {
    let operList = ["+-*", "+*-", "*-+", "*+-", "-+*", "-*+"];
    let answer = 0;
    for (let i = 0; i < 6; i++) {
        let tmp = foo(operList[i], expression);
        if (answer < tmp) {
            answer = tmp;
        }
    }
    return answer;
}

File: maximizeExpression.cpp
#include <bits/stdc++.h>
using namespace std;
using ll = long long;

int check[3];
char oper[3] = {'+', '-', '*'};
vector<int> bt;
vector<string> a;
ll ans;
void foo() {
    if (bt.size() == 3) {
        auto b = a;
        for (auto bte : bt) {
            char op = oper[bte];
            for (int i = 0; i < b.size(); i++) {
                if (b[i][0] == op) {
                    if (op == '+') {
                        b[i - 1] = to_string(stoll(b[i - 1]) + stoll(b[i + 1]));
                    } else if (op == '-') {
                        b[i - 1] = to_string(stoll(b[i - 1]) - stoll(b[i + 1]));
                    } else if (op == '*') {
                        b[i - 1] = to_string(stoll(b[i - 1]) * stoll(b[i + 1]));
                    }
                    b.erase(b.begin() + i);
                    b.erase(b.begin() + i);
                    i--;
                }
            }
        }
        ans = max(ans, abs(stoll(b[0])));
        return;
    }
    for (int i = 0; i < 3; i++) {
        if (check[i] == 0) {
            check[i] = 1;
            bt.push_back(i);
            foo();
            bt.pop_back();
            check[i] = 0;
        }
    }
}

ll solution(string exp) {
    stringstream f(exp);
    string s = "";
    for (int i = 0; i < exp.size(); i++) {
        if (exp[i] == '+' || exp[i] == '-' || exp[i] == '*') {
            a.push_back(s);
            string tmp = "";
            tmp += exp[i];
            a.push_back(tmp);
            s = "";
        } else {
            s += exp[i];
        }
    }
    a.push_back(s);
    foo();
    return ans;
}